convert.py

```diff
@@ -738,6 +743,9 @@ def add_meta_arch(self, params: Params) -> None:
         self.gguf.add_head_count_kv       (params.n_head_kv)
         self.gguf.add_layer_norm_rms_eps  (params.f_norm_eps)
 
+        if params.ftype:
+            self.gguf.add_file_type(params.ftype)
```

Comment:
Will need some help with Python - it is very difficult to write code without type checking

Not sure why I get this error:

```java
Writing models/7B/ggml-model-f16.gguf, format GGMLFileType.MostlyF16
Traceback (most recent call last):
  File "/Users/ggerganov/development/github/llama.cpp/convert.py", line 1071, in <module>
    main()
  File "/Users/ggerganov/development/github/llama.cpp/convert.py", line 1066, in main
    OutputFile.write_all(outfile, params, model, vocab)
  File "/Users/ggerganov/development/github/llama.cpp/convert.py", line 803, in write_all
    of.add_meta_arch(params)
  File "/Users/ggerganov/development/github/llama.cpp/convert.py", line 747, in add_meta_arch
    self.gguf.add_file_type(params.ftype)
  File "/Users/ggerganov/development/github/llama.cpp/gguf.py", line 600, in add_file_type
    self.add_uint32(KEY_GENERAL_FILE_TYPE, ftype)
  File "/Users/ggerganov/development/github/llama.cpp/gguf.py", line 435, in add_uint32
    self.add_val(val, GGUFValueType.UINT32)
  File "/Users/ggerganov/development/github/llama.cpp/gguf.py", line 479, in add_val
    self.kv_data += struct.pack("<I", val)
                    ^^^^^^^^^^^^^^^^^^^^^^
struct.error: required argument is not an integer
```

Reply:
We can also add a string - I'm trying to restore the pre-GGUF behavior of `ftype`